crates/extensions/tedge_mqtt_ext/src/lib.rs

```diff
@@ -145,19 +145,19 @@ impl MqttMessageBox {
         }
     }
 
-    fn split(&mut self) -> (FromPeers<'_>, ToPeers<'_>) {
+    fn split(self) -> (FromPeers, ToPeers) {
```

Comment:
Both `fn slip` and `struct MqttMessageBox` can be deprecated as built & split only here.

Reply:
I don't quite understand what you mean here. I introduced `split` as a mechanism to enable concurrent processing of incoming and outgoing messages, without the need for other actors to know/care about this detail (so outside there is still a single message box for a single actor). `MqttActor` could potentially be split into two actors to deal with this, but that feels like it's just going to increase complexity in all the MQTT interactions for other actors.

Reply:
I meant that `MqttActor` doesn't need to define a message box as a single object. In fact no actor has to. The abstraction with other actors is implemented by the builder with appropriate  `ServiceProvider` / `MessageSource` implementation.

The following is working fine:

```
pub struct MqttActor {
    mqtt_config: mqtt_channel::Config,
    from_peers: FromPeers,
    to_peers: ToPeers,
}
```

Reply:
I pushed a fixup commit: https://github.com/thin-edge/thin-edge.io/pull/2329/commits/cc067c1220c0d696d8d4ceb27dac6ac08e8adaf9